adapters/sidekiq/lib/opentelemetry/adapters/sidekiq/adapter.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Sidekiq
+      # The Adapter class contains logic to detect and install the Sidekiq
+      # instrumentation adapter
+      class Adapter < OpenTelemetry::Instrumentation::Adapter
+        install do |_config|
+          require_dependencies
+          add_client_middleware
+          add_server_middleware
+        end
+
+        present do
+          defined?(::Sidekiq)
+        end
+
+        private
+
+        def require_dependencies
+          require_relative 'middlewares/client/tracer_middleware'
+          require_relative 'middlewares/server/tracer_middleware'
+        end
+
+        def add_client_middleware
+          ::Sidekiq.configure_client do |config|
+            config.client_middleware do |chain|
+              chain.add Middlewares::Client::TracerMiddleware
+            end
+          end
+        end
+
+        def add_server_middleware
+          ::Sidekiq.configure_server do |config|
+            config.server_middleware do |chain|
+              chain.add Middlewares::Server::TracerMiddleware
+            end
+          end
+
+          ::Sidekiq.configure_client do |config|
+            config.client_middleware do |chain|
+              chain.add Middlewares::Client::TracerMiddleware
+            end
+          end
+
+          if defined?(::Sidekiq::Testing)
```

Comment:
This is required to be able to actually test server middleware, it's also beneficial to make it part of the installation as people will be including this in their apps, and having this available by default seems like a reasonable behaviour.